### PR TITLE
GH-43873: [Go][CI] Remove Go related test CI

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

The Go implementation is moving to apache/arrow-go from go/ in apache/arrow.

### What changes are included in this PR?

CI has been migrated to apache/arrow-go repository.

### Are these changes tested?

No

### Are there any user-facing changes?

No
* GitHub Issue: #43873

### CODE REVIEW

**Comment by raulcd** [comment]
This PR should be merged after: https://github.com/apache/arrow/pull/44142 in order to fix the integration error as we don't require the local `go_build.sh` script anymore after those.

File: .env [hunk 1]
```diff
@@ -58,7 +58,6 @@ CUDA=11.2.2
 DASK=latest
 DOTNET=8.0
 GCC_VERSION=""
-GO=1.22.6
 STATICCHECK=v0.5.1
```
Comment (kou) [inline .env:61]: Could you also remove this?

File: docker-compose.yml [hunk 2]
```diff
@@ -1866,7 +1801,6 @@ services:
         jdk: 17
         maven: ${MAVEN}
         node: ${NODE}
-        go: ${GO}
```
Comment (kou) [inline docker-compose.yml:1869]: Can we always use the latest Go in `ci/docker/conda-integration.dockerfile`?

https://github.com/apache/arrow-go/blob/12482ca3eb1cd4da21c45325e174f1dac871281b/dev/release/verify_rc.sh#L137-L156 may help us.

Comment (raulcd) [reply docker-compose.yml:1869]: I've tried something here: https://github.com/apache/arrow/pull/44143/commits/3d2516dec16dccf542cdc610a5ddf2ce9d282cf8

**Review by kou — APPROVED** [review]
+1

**Pull request merged**